src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -6595,7 +6595,7 @@ private void ProcessSSPI(int receivedLength)
             SniContext outerContext = _physicalStateObj.SniContext;
             _physicalStateObj.SniContext = SniContext.Snix_ProcessSspi;
             // allocate received buffer based on length from SSPI message
-            byte[] receivedBuff = new byte[receivedLength];
+            byte[] receivedBuff = ArrayPool<byte>.Shared.Rent(receivedLength);
```

Comment:
I assume `throw SQL.SynchronousCallMayNotPend(); ` is not commonly taken, otherwise we should return the array in that path.

Reply:
It's a real error if it occurs, it means someone has tried to do asynchronous things in a synchronous context. It's a programming error if that happens and It'll blow up the unit tests all over the place.